packages/synapse-react-client/src/components/Authentication/LastLoginInfo.tsx

```diff
@@ -0,0 +1,199 @@
+  LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+} from '../../utils/SynapseConstants'
+import { LoginMethod, getLoginMethodFriendlyName } from './LoginMethod'
+
+export type LastLoginInfoProps = {
+  currentSourceAppName: string | null
+  lastLoginMethod: LoginMethod | null
+  lastLoginSourceAppName: string | null
+  lastLoginSourceAppURL: string | null
+  lastLoginDate: string | null
+  display: 'sentence' | 'box'
+}
+
+export function getLastLoginInfo(): Omit<LastLoginInfoProps, 'display'> {
+  return {
+    currentSourceAppName: localStorage.getItem(
+      CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginMethod: localStorage.getItem(
+      LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    ) as LoginMethod,
+    lastLoginSourceAppName: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginSourceAppURL: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    ),
+    lastLoginDate: localStorage.getItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY),
+  }
+}
+
+export function setCurrentAppInfo(currentSourceAppName: string) {
+  localStorage.setItem(
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    currentSourceAppName,
+  )
+}
+
+export function setLastLoginInfo(
+  lastLoginMethod: LoginMethod,
+  lastLoginDate: string,
+  lastLoginSourceAppName: string,
+  lastLoginSourceAppURL: string,
+) {
+  localStorage.setItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY, lastLoginDate)
+  localStorage.setItem(LAST_LOGIN_METHOD_LOCALSTORAGE_KEY, lastLoginMethod)
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    lastLoginSourceAppName,
+  )
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    lastLoginSourceAppURL,
+  )
+}
+
+export function clearLastLoginInfo() {
+  const keys = [
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  ]
+  keys.forEach(key => localStorage.removeItem(key))
+}
+
+function getSourceAppIndefiniteArticle(val: string) {
+  return val.match(/^[aeiou]/i) ? 'an' : 'a'
+}
+
+function getSourceAppDefiniteArticle(val: string) {
+  return val.match(' Portal$') ? 'the ' : ''
+}
+
+export default function LastLoginInfo(props: LastLoginInfoProps) {
+  const {
+    currentSourceAppName,
+    lastLoginMethod,
+    lastLoginSourceAppName,
+    lastLoginSourceAppURL,
+    lastLoginDate,
+    display,
+  } = props
+
+  const shouldNotShowLastLoginInfo =
+    currentSourceAppName === null ||
+    lastLoginMethod === null ||
+    lastLoginSourceAppName === null ||
+    lastLoginSourceAppURL === null ||
+    lastLoginDate === null ||
+    currentSourceAppName === lastLoginSourceAppName
+
+  if (shouldNotShowLastLoginInfo) {
+    return null
+  }
```

Comment:
Don't render the last login info if any of the local storage values are missing or if the current portal matches the previous portal logged into by the user